.github/workflows/release.yml

```diff
@@ -95,15 +150,12 @@ jobs:
       - name: go-check
         run: go version
 
-      - name: Node setup
-        uses: actions/setup-node@39370e3970a6d050c480ffad4ff0ed4d3fdee5af # v4.1.0
+      - name: UI Cache Setup
+        uses: actions/cache@v4
         with:
-          node-version-file: "./ui/package.json"
-
-      - name: Yarn setup
-        id: install-yarn
-        run: |
-          npm install -g yarn
+          path: |
+            http/web_ui/
+          key: ${{ github.ref }}-ui
```

Comment:
This will mean for actual releases (e.g., `v2.2.0`), if the UI fails to build and we need to patch it, we'll have to bump the minor version I believe?

This should be fine, but probably warrants a follow-up note in our checklist.

Reply:
If the UI fails to build, the rest of the pipeline shouldn't run